### PR TITLE
LLM: improve gpu supports key feature doc page

### PR DESCRIPTION
**Pull request by plusbang**
## Description

Improvements on gpu supports key feature doc page:
- separate into two pages: Inference on GPU, Finetune (QLora)
- introduce how to run LLMs on gpu in detail and highlight differences with operations on cpu
- add qlora finetuning part

Besides, add Intel Data Center GPU Max Series in installation page

### 4. How to test?
- [ ] Document test (https://binbin-doc.readthedocs.io/en/improve-doc-v5/doc/LLM/Overview/KeyFeatures/gpu_supports.html)

### CODE REVIEW

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 1]
```diff
@@ -1,47 +1,166 @@
...
+
+         saved_dir='./llama-2-bigdl-llm-4-bit'
+         model = AutoModelForCausalLM.load_low_bit(saved_dir) # Load the optimized model
+
+         model = model.to('xpu') # Important after obtaining the optimized model
 ```
 
+### Run Optimized Model
+
+You could then inference using the optimized model on Intel GPUs almostly the same as on CPUs. **The only difference is to set `to('xpu')` for input tensors and `.cpu()` for output tensors.**
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:87]: I guess "inference" is not a verb? How about "do inference"?

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 2]
```diff
@@ -1,47 +1,166 @@
...
+   output = model.generate(input_ids, max_new_tokens=32)
+   torch.xpu.synchronize()
+   output = output.cpu() # Move to CPU
+   output_str = tokenizer.decode(output[0], skip_special_tokens=True)
+```
+
+```eval_rst
+.. note::
+
+   The initial generation of optimized LLMs on Intel GPUs could be slow. Therefore, it's advisable to perform a **warm-up** run before the actual generation.
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:105]: How about "it's recommended"?

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 3]
```diff
@@ -1,47 +1,166 @@
...
 ```
 
-After loading the Hugging Face *Transformers* model, you may easily run the optimized model as follows:
+First, load model using `transformers`-style API and **set it to `to('xpu')`** as follows:
+```python
+import intel_extension_for_pytorch as ipex
+from bigdl.llm.transformers import AutoModelForCausalLM
+
+model = AutoModelForCausalLM.from_pretrained("meta-llama/Llama-2-7b-hf",
+                                             load_in_low_bit="nf4",
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:133]: We use "nf4" here according to the QLoRA paper (https://arxiv.org/pdf/2305.14314.pdf), which is said to be able to yield better model quality than "int4".

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 4]
```diff
@@ -1,47 +1,166 @@
...
+                                             torch_dtype=torch.float16,
+                                             modules_to_not_convert=["lm_head"],)
+model = model.to('xpu')
+```
 
+Then, we have to apply some preprocessing to the model to prepare it for training.
 ```python
-# run the optimized model
-from transformers import AutoTokenizer
+from bigdl.llm.transformers.qlora import prepare_model_for_kbit_training
```
Comment (yangw1234) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:142]: Instead of import `prepare_model_for_kbit_training` and `get_peft_model` from `peft`, as we did for regular qlora using bitandbytes and cuda. Here we import them from `bigdl.llm.transformers.qlora` to get a BigDL-LLM compatible peft model.

Comment (yangw1234) [reply docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:142]: The rest of the code is same as regular LoRA finetuning using "peft"

Comment (plusbang) [reply docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:143]: > Instead of import `prepare_model_for_kbit_training` and `get_peft_model` from `peft`, as we did for regular qlora using bitandbytes and cuda. Here we import them from `bigdl.llm.transformers.qlora` to get a BigDL-LLM compatible peft model.

Have added related descriptions.

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 5]
```diff
@@ -1,47 +1,171 @@
...
 
-# load Hugging Face Transformers model with INT4 optimizations on Intel GPUs
-from bigdl.llm.transformers import AutoModelForCausalLM
+### Load and Optimize Model
 
-model = AutoModelForCausalLM.from_pretrained('/path/to/model/',
-                                             load_in_4bit=True,
-                                             optimize_model=False)
-model = model.to('xpu')
+You could choose to use [PyTorch API](./optimize_model.html) or [`transformers`-style API](./transformers_style_api.html) on Intel GPUs based on your requirements.
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:17]: based on your requirements => according to your preference ?

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 6]
```diff
@@ -1,47 +1,171 @@
...
+         import intel_extension_for_pytorch as ipex
+         from transformers import LlamaForCausalLM
+         from bigdl.llm import optimize_model
+
+         model = LlamaForCausalLM.from_pretrained('meta-llama/Llama-2-7b-chat-hf', torch_dtype='auto', low_cpu_mem_usage=True)
+         model = optimize_model(model) # With only one line to enable BigDL-LLM INT4 optimization
+
+         model = model.to('xpu') # Important after obtaining the optimized model
+
+      Especially, if you have saved the optimized model following setps `here <https://bigdl.readthedocs.io/en/latest/doc/LLM/Overview/KeyFeatures/optimize_model.html#save>`_, the loading process on Intel GPUs maybe as follows:
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:40]: Please use relative link here to help with version control :D

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 7]
```diff
@@ -1,47 +1,171 @@
...
+         # Take Llama-2-7b-chat-hf as an example
+         import intel_extension_for_pytorch as ipex
+         from bigdl.llm.transformers import AutoModelForCausalLM
+
+         # Load model in 4 bit, which convert the relevant layers in the model into INT4 format
+         model = AutoModelForCausalLM.from_pretrained('meta-llama/Llama-2-7b-chat-hf', load_in_4bit=True)
+
+         model = model.to('xpu') # Important after obtaining the optimized model
+
+      Especially, if you have saved the optimized model following setps `here <https://bigdl.readthedocs.io/en/latest/doc/LLM/Overview/KeyFeatures/hugging_face_format.html#save-load>`_, the loading process on Intel GPUs maybe as follows:
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:72]: Same here

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 8]
```diff
@@ -1,47 +1,171 @@
...
+
+Continuing with the [example of Llama-2-7b-chat-hf](#load-and-optimize-model), running as follows:
+```python
+import torch
+
+with torch.inference_mode():
+   prompt = 'Q: What is CPU?\nA:'
+   input_ids = tokenizer.encode(prompt, return_tensors="pt").to('xpu') # With .to('xpu') specifically for inference on Intel GPUs
+   output = model.generate(input_ids, max_new_tokens=32)
+   torch.xpu.synchronize()
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:97]: `torch.xpu.synchronize()`

should we remove this as it is not necessary?

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 9]
```diff
@@ -1,47 +1,171 @@
...
 .. note::
 
-   You may apply INT8 optimizations as follows:
+   The initial generation of optimized LLMs on Intel GPUs could be slow. Therefore, it's recommended to perform a **warm-up** run before the actual generation.
+```
+
+
+## QLoRA Finetuning
+
+We also support finetuning LLMs (large language models) using Q-Lora with BigDL-LLM 4bit optimizations on Intel GPUs.
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:111]: Q-Lora -> QLoRA? We may keep one for consistency

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 10]
```diff
@@ -1,47 +1,171 @@
...
+
+```eval_rst
+.. note::
+
+   Currently, only Hugging Face Transformers models are supported running Q-Lora finetuning.
```
Comment (Oscilloscope98) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:116]: Same here

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 11]
```diff
@@ -1,47 +1,170 @@
...
+
+Continuing with the [example of Llama-2-7b-chat-hf](#load-and-optimize-model), running as follows:
+```python
+import torch
+
+with torch.inference_mode():
+   prompt = 'Q: What is CPU?\nA:'
+   input_ids = tokenizer.encode(prompt, return_tensors="pt").to('xpu') # With .to('xpu') specifically for inference on Intel GPUs
+   output = model.generate(input_ids, max_new_tokens=32)
+   output = output.cpu() # Move to CPU
```
Comment (shane-huang) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:97]: I recall that in some situations `output=output.cpu()` is not needed? anyway it does no harm to add this line, right?

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md [hunk 12]
```diff
@@ -1,47 +1,170 @@
...
 
 ```eval_rst
 .. note::
 
-   You may apply INT8 optimizations as follows:
+   The initial generation of optimized LLMs on Intel GPUs could be slow. Therefore, it's recommended to perform a **warm-up** run before the actual generation.
+```
+
+
+## QLoRA Finetuning
```
Comment (shane-huang) [inline docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md:108]: Maybe separate QLora Finetuning and inference into Two pages. e.g. 
- Inference on GPU
- Finetune (QLora)

File: docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md (hunk 11 above)
Comment (plusbang) [reply docs/readthedocs/source/doc/LLM/Overview/KeyFeatures/gpu_supports.md]: Have separated into two pages. 

> I recall that in some situations `output=output.cpu()` is not needed? anyway it does no harm to add this line, right?

This line may cause errors sometimes (for example, if we run bark model through TTS, the output maybe `numpy.ndarray`, then `.cpu()` may cause error). We only use it to move tensors to CPU, and it is not a necessary operation, so I remove it now.

**Review by shane-huang — APPROVED** [review]

**Comment by shane-huang** [comment]
LGTM
